synapse/rest/client/v1/room.py

```diff
@@ -602,6 +604,9 @@ async def on_POST(self, request, room_id):
                 "prev_events": prev_event_ids.copy(),
             }
 
+            # Mark all events as historical
+            event_dict["content"][EventContentFields.MSC2716_HISTORICAL] = True
```

Comment:
Just moved this below to make it consistent with the way we modify the state events above.